libc/test/src/sys/mman/linux/mincore_test.cpp

```diff
@@ -0,0 +1,125 @@
+  EXPECT_THAT(LIBC_NAMESPACE::munmap(addr, page_size), Succeeds());
+}
+
+TEST(LlvmLibcMincoreTest, InvalidVec) {
+  unsigned long page_size = LIBC_NAMESPACE::sysconf(_SC_PAGESIZE);
+  void *addr = LIBC_NAMESPACE::mmap(nullptr, 4 * page_size, PROT_READ,
+                                    MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+  EXPECT_NE(addr, MAP_FAILED);
+  EXPECT_EQ(reinterpret_cast<unsigned long>(addr) % page_size, 0ul);
+  libc_errno = 0;
+  int res = LIBC_NAMESPACE::mincore(addr, 1, nullptr);
+  EXPECT_THAT(res, Fails(EFAULT, -1));
+  void *area = LIBC_NAMESPACE::mmap(nullptr, page_size, PROT_READ | PROT_WRITE,
+                                    MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+  EXPECT_NE(area, MAP_FAILED);
+  unsigned char *ptr = static_cast<unsigned char *>(area) + page_size - 3;
+  res = LIBC_NAMESPACE::mincore(addr, 4 * page_size, ptr);
+  EXPECT_THAT(res, Fails(EFAULT, -1));
+  EXPECT_THAT(LIBC_NAMESPACE::munmap(addr, page_size), Succeeds());
+  EXPECT_THAT(LIBC_NAMESPACE::munmap(area, 2), Succeeds());
+}
+
+TEST(LlvmLibcMincoreTest, NoError) {
+  unsigned long page_size = LIBC_NAMESPACE::sysconf(_SC_PAGESIZE);
+  void *addr = LIBC_NAMESPACE::mmap(nullptr, page_size, PROT_READ,
+                                    MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+  EXPECT_NE(addr, MAP_FAILED);
+  EXPECT_EQ(reinterpret_cast<unsigned long>(addr) % page_size, 0ul);
+  unsigned char vec;
+  libc_errno = 0;
+  int res = LIBC_NAMESPACE::mincore(addr, 1, &vec);
+  EXPECT_THAT(res, Succeeds());
+  EXPECT_THAT(LIBC_NAMESPACE::munmap(addr, page_size), Succeeds());
+}
+
+TEST(LlvmLibcMincoreTest, NegativeLength) {
+  unsigned long page_size = LIBC_NAMESPACE::sysconf(_SC_PAGESIZE);
+  void *addr = LIBC_NAMESPACE::mmap(nullptr, page_size, PROT_READ,
+                                    MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+  EXPECT_NE(addr, MAP_FAILED);
+  EXPECT_EQ(reinterpret_cast<unsigned long>(addr) % page_size, 0ul);
+  unsigned char vec;
+  libc_errno = 0;
+  int res = LIBC_NAMESPACE::mincore(addr, -1, &vec);
+  EXPECT_THAT(res, Fails(ENOMEM, -1));
+  EXPECT_THAT(LIBC_NAMESPACE::munmap(addr, page_size), Succeeds());
+}
+
+TEST(LlvmLibcMincoreTest, PageOut) {
+  unsigned long page_size = LIBC_NAMESPACE::sysconf(_SC_PAGESIZE);
+  unsigned char vec;
+  void *addr = LIBC_NAMESPACE::mmap(nullptr, page_size, PROT_READ | PROT_WRITE,
+                                    MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+  EXPECT_NE(addr, MAP_FAILED);
+  EXPECT_EQ(reinterpret_cast<unsigned long>(addr) % page_size, 0ul);
+
+  // touch the page
+  {
+    static_cast<char *>(addr)[0] = 0;
+    LIBC_NAMESPACE::syscall_impl(
+        SYS_mlock, reinterpret_cast<unsigned long>(addr), page_size);
+    libc_errno = 0;
+    int res = LIBC_NAMESPACE::mincore(addr, 1, &vec);
+    EXPECT_EQ(vec & 1u, 1u);
+    EXPECT_THAT(res, Succeeds());
+    LIBC_NAMESPACE::syscall_impl(
+        SYS_munlock, reinterpret_cast<unsigned long>(addr), page_size);
```

Comment:
Let's add support for mlock+munlock, then use those syscall wrappers here (similar to what you did for `sysconf`), rather than open coding the syscalls in this fashion. Else add a TODO in a comment.